contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >= 0.7.0 < 0.9.0;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENSRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@ew-did-registry/proxyidentity/contracts/IOwned.sol";
+import "./RoleDefinitionResolver.sol";
+import {ClaimManager} from "./ClaimManager.sol";
+
+interface EthereumDIDRegistry {
+    function identityOwner(address identity) external view returns(address);
+    function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool);
+}
+    
+contract RevocationRegistryOnChain {
+    bytes32 constant private ASSERTION_DELEGATE_TYPE = 0x766572694b657900000000000000000000000000000000000000000000000000;
+    string constant private ERC712_DOMAIN_NAME = "Revocation Registry";
+    string constant private ERC712_DOMAIN_VERSION = "1.0";
+
+    struct RevokedClaim {
+        address revoker;
+        uint revokedTimestamp;
+        bytes32 revokerRoleDigest; //namehash
+    }
+
+    mapping(bytes32 => RevokedClaim) revokedClaims;
+    address private didRegistry;
+    address private ensRegistry;
+    address private claimManager;
+    
+    event Revoked(address revoker, bytes32 claimDigest);
+
+    constructor(address _didRegistry, address _ensRegistry, address _claimManager){
+        didRegistry = _didRegistry;
+        ensRegistry = _ensRegistry;
+        claimManager = _claimManager;
+    }
+
+    function revokeClaim(bytes32 claimDigest, bytes32 role, address revoker, bytes32 revokerRoleDigest) public {
+        require(revokedClaims[claimDigest].revokedTimestamp == 0, "The claim is already revoked");
+        verifyRevoker(revoker, role, revokerRoleDigest);
+        revokedClaims[claimDigest].revoker = revoker;
+        revokedClaims[claimDigest].revokedTimestamp = block.number;
+        revokedClaims[claimDigest].revokerRoleDigest = revokerRoleDigest;
+        emit Revoked(revoker, claimDigest);
+    }
+
+    function revokeClaimInList(bytes32 [] memory claimList,bytes32 role, address revoker, bytes32 revokerRoleDigest) public{
```

Comment:
If I have a good understanding of the purpose of this function, you revoke all claims present in a list.
            What if we call this revokeClaimsInList instead ? Otherwise one could think you lookup a specific claim in a list of claims and just revoke this claim
```suggestion
    function revokeClaimsInList(bytes32 [] memory claimList,bytes32 role, address revoker, bytes32 revokerRoleDigest) public{
```

Reply:
Agreed. I would change that, thanks.